src/coreclr/vm/dynamicmethod.cpp

```diff
@@ -954,6 +956,19 @@ void LCGMethodResolver::RecycleIndCells()
             cellcurr = list->indcell;
             _ASSERTE(cellcurr != NULL);
 
+#if defined (FEATURE_CACHED_INTERFACE_DISPATCH)
+            // Cached dispatch dispatch uses dynamically allocated caches that need to be freed individually
+            if (UseCachedInterfaceDispatch())
+            {
+                InterfaceDispatchCell *pDispatchCell = (InterfaceDispatchCell*)cellcurr;
+                InterfaceDispatchCacheHeader* cellCacheHeader = pDispatchCell->GetCache();
+                if (cellCacheHeader != NULL)
+                {
+                    InterfaceDispatch_DiscardCacheHeader(cellCacheHeader);
+                    pDispatchCell->m_pCache = 0;
```

Comment:
This is odd. We have a getter, `GetCache()`, above, but then we unilaterally manipulate the field. We should either always access the field or do it through methods/abstractions.